tests/unit/test_mysql.py

```diff
@@ -115,34 +119,38 @@ def test_does_mysql_user_exist_failure(self, _run_mysqlcli_script):
         with self.assertRaises(MySQLCheckUserExistenceError):
             self.mysql.does_mysql_user_exist("test_username", "1.1.1.1")
 
-    @patch("charms.mysql.v0.mysql.MySQLBase._run_mysqlcli_script")
-    def test_configure_mysqlrouter_user(self, _run_mysqlcli_script):
+    @patch("charms.mysql.v0.mysql.MySQLBase.get_cluster_primary_address")
+    @patch("charms.mysql.v0.mysql.MySQLBase._run_mysqlsh_script")
```

Comment:
Since you are _always_ going to need to patch `_run_mysqlsh_script`, it could be made a pytest fixture with `autouse=True`